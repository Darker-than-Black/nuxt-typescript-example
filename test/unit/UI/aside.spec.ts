import VueTestInstance from '~/test/factory/vue-instance'
import Aside from '~/components/UI/aside.vue'

describe('aside.vue', () => {
  const navLinks: any = [
    { text: 'Загальне', icon: 'Home', path: '/' }
  ]
  const wrapper = VueTestInstance(Aside, {
    options: {
      data: () => ({ navLinks }),
      props: { simple: false }
    }
  })

  it('initialize correctly', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('check show link text', () => {
    const link = wrapper.find('.aside .aside__item:nth-child(1)')
    expect(link.find('.aside__text').text()).toBe(navLinks[0].text)
  })

  it('check path to icon', () => {
    expect(wrapper.find('.aside .aside__item img').attributes().src).toBe(`/icons/${navLinks[0].icon}.svg`)
  })

  it('check update prop "simple"', () => {
    wrapper.setProps({ simple: true })
    expect(wrapper.props().simple).toBe(true)
  })

  it('check hide link text', () => {
    expect(wrapper.find('.aside .aside__item .aside__text').classes('d-none')).toBe(true)
  })
})
