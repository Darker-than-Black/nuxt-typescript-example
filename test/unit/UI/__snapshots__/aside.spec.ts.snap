// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`aside.vue should match the snapshot 1`] = `
<ul class="aside">
  <li class="aside__item"><img src="/icons/Home.svg" alt="Загальне" class="img mr-2"> <span class="aside__text">
      Загальне
    </span></li>
</ul>
`;
